packages/flutter_tools/test/commands/daemon_test.dart

```diff
@@ -103,9 +103,9 @@ void main() {
         notifyingLogger: notifyingLogger
       );
       commands.add(<String, dynamic>{'id': 0, 'method': 'daemon.shutdown'});
-      return daemon.onExit.then<Null>((int code) {
-        responses.close();
-        commands.close();
+      return daemon.onExit.then<Null>((int code) async {
+        await responses.close();
```

Comment:
@devoncarew I couldn't understand this test enough to figure out why this hangs. Do you have any clues?

Reply:
ok, removed